### ISSUE TITLE
FlowComp does not define flow / compilation server (heaps domkit sample)

### ISSUE THREAD

**Issue by jcward** [ncannasse/domkit#10]
Testing domkit in heaps / samples / domkit -- I get the following error on the 2nd compile after a small change in sample's `class Domkit`, unless I restart the compilation server:

```
> Executing task: haxe --connect 6000 domkit_hl.hxml <

/home/jward/dev/domkit/domkit/Macros.hx:391: characters 3-8 : Uncaught exception UIKitError(h2d.domkit.FlowComp does not define component flow) 0
/home/jward/dev/domkit/domkit/Macros.hx:85: characters 4-78 : Called from here
/home/jward/dev/domkit/domkit/MetaComponent.hx:48: characters 38-69 : Called from here
/home/jward/dev/domkit/domkit/Macros.hx:360: characters 13-62 : Called from here
/home/jward/dev/heaps/h2d/domkit/InitComponents.hx:18: characters 10-37 : Called from here
./Domkit.hx:62: character 1 : Called from here
The terminal process command 'haxe --connect 6000 domkit_hl.hxml' failed to launch (exit code: 1)
```

Heaps and domkit are both git / master. This is against Haxe 4.0.0-rc3.

I tried a Haxe nightly from today, but the compilation server wouldn't start, with:

```
/home/jward/dev/heaps/hxsl/MacroParser.hx:170: characters 12-16 : haxe.macro.FunctionKind should be String
/home/jward/dev/heaps/hxsl/MacroParser.hx:170: characters 12-16 : For function argument 'f'
/home/jward/dev/heaps/hxsl/MacroParser.hx:168: lines 168-185 : Void should be hxsl.ExprDef
/home/jward/dev/heaps/hxsl/BatchShader.hx:3: character 1 : Build failure
```

**Comment by ncannasse** [comment]
Yes, this requires a quite recent haxe, but latest development has changed a few things that I need to fix first.

**Comment by manspaniel** [comment]
Hi @ncannasse, love your work!

I'd love to dive into domkit, it looks great. Is there a haxe/heaps/domkit version combo which should work well together?

Cheers

**Comment by ncannasse** [comment]
This issue was closed with d1095a926a46b03dccb7da903340a107beae4dc2

**Comment by ncannasse** [comment]
@manspaniel you can check https://heaps.io/documentation/domkit.html
